jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/milestones/CaseMilestoneItemView.java

```diff
@@ -0,0 +1,127 @@
+import org.jbpm.console.ng.cm.client.util.AbstractView;
+
+import org.jbpm.console.ng.cm.client.util.CaseMilestoneStatus;
+import org.jbpm.console.ng.cm.client.util.DateConverter;
+import org.jbpm.console.ng.cm.model.CaseMilestoneSummary;
+
+import static org.jboss.errai.common.client.dom.DOMUtil.*;
+
+@Dependent
+@Templated
+public class CaseMilestoneItemView extends AbstractView<CaseMilestoneListPresenter> implements TakesValue<CaseMilestoneSummary>, IsElement {
+
+    @Inject
+    @DataField("milestone-name")
+    @Bound
+    Span name;
+
+    @Inject
+    @DataField("milestone-status")
+    Span status;
+
+    @Inject
+    @DataField("list-group-item")
+    Div listGroupItem;
+    
+    @Inject
+    @AutoBound
+    private DataBinder<CaseMilestoneSummary> caseMilestoneSummary;
+
+    @Inject
+    protected TranslationService translationService;
+
+    @PostConstruct
+    public void init() {
+        tooltip(status);
+    }
+    @Override
+    public HTMLElement getElement() {
+        return listGroupItem;
+    }
+
+    @Override
+    public CaseMilestoneSummary getValue() {
+        return caseMilestoneSummary.getModel();
+    }
+
+    @Override
+    public void setValue(final CaseMilestoneSummary model) {
+        this.caseMilestoneSummary.setModel(model);
+        final CaseMilestoneStatus milestoneStatus = CaseMilestoneStatus.fromStatus(model.getStatus());
+
+        String statusStr = convertStatusToStr(model.getStatus());
+
+        switch(milestoneStatus){
+            case AVAILABLE: {
+                showMilestoneStatus( statusStr, statusStr, "label", "label-info");
+                break;
+            }
+            case COMPLETED:{
+                String achievedAtStr  = DateConverter.getDateStr(model.getAchievedAt());
+                showMilestoneStatus( "", statusStr + (achievedAtStr.isEmpty() ? "" : "("+ achievedAtStr +")"), "pficon", "pficon-ok");
+                break;
+            }
+            case TERMINATED:{
+                String achievedAtStr  = DateConverter.getDateStr(model.getAchievedAt());
+                showMilestoneStatus( "", statusStr + (achievedAtStr.isEmpty() ? "" : "("+ achievedAtStr +")"), "pficon", "pficon-error-circle-o");
+                break;
+            }
+        }
+    }
+
+    public void showMilestoneStatus(final String statusText, final String tooltipTitle,final String ... stylesClass) {
+        status.setTextContent(statusText);
+        status.setAttribute("data-original-title", tooltipTitle);
+        tooltip(status);
+        for(String styleClass :stylesClass){
+            addCSSClass(this.status, styleClass);
+        }
+    }
+
+    public String convertStatusToStr(final String modelValue){
+        if (modelValue == null) {
+            return "";
+        } else {
+            return translationService.format(CaseMilestoneStatus.fromStatus(modelValue).getStatus());
+        }
+    }
+
+    private native void tooltip(final HTMLElement e) /*-{
```

Comment:
Perhaps move this into AbstractView class so it can be shared with CaseInstanceViewImpl.